beta/nncf/tensorflow/layers/wrapper.py

```diff
@@ -65,14 +66,75 @@ def trainable(self):
+                op.trainable = value
 
     @property
     def trainable_weights(self):
-        return self._trainable_weights + self.layer.trainable_weights
+        trainable_weights_by_definition = self._trainable_weights + self.layer.trainable_weights
+
+        if self._op_build:
+            trainable_ops_weights_on, trainable_ops_weights_off = self.classify_trainable_ops_weights()
+        else:
+            trainable_ops_weights_on, trainable_ops_weights_off = [], []
+
+        if self.trainable:
+            weights = trainable_weights_by_definition
+            for trainable_weight_off in trainable_ops_weights_off:
+                weights.remove(trainable_weight_off)
+            outputs = weights
+        else:
+            outputs = trainable_ops_weights_on
+        return outputs
 
     @property
     def non_trainable_weights(self):
-        return self._non_trainable_weights + self.layer.non_trainable_weights
+        trainable_weights_by_definition = self._trainable_weights + self.layer.trainable_weights
+        non_trainable_weights_by_definition = self._non_trainable_weights + self.layer.non_trainable_weights
+
+        if self._op_build:
+            trainable_ops_weights_on, trainable_ops_weights_off = self.classify_trainable_ops_weights()
+        else:
+            trainable_ops_weights_on, trainable_ops_weights_off = [], []
+
+        if self.trainable:
+            outputs = non_trainable_weights_by_definition + trainable_ops_weights_off
+        else:
+            weights = trainable_weights_by_definition
+            for trainable_weight_on in trainable_ops_weights_on:
+                weights.remove(trainable_weight_on)
+            outputs = non_trainable_weights_by_definition + weights
+        return outputs
+
+    def classify_trainable_ops_weights(self):
+        """
+        Classifies operation trainable weights by trainable property of corresponding operation.
+        Note: Operation with trainable weights can be switched off from training.
+              Operation trainable property can be adjusted
+              to be different from NNCFWrapper global trainable property.
+
+        :return trainable_ops_weights_on: list of trainable operation weights
+                which is getting updated during training.
+        :return trainable_ops_weights_off: list of trainable operation weights
+                which is NOT getting updated during training.
+        """
+        trainable_ops_weights_on = []
+        trainable_ops_weights_off = []
+        for ops in self.weights_attr_ops.values():
+            for op_name, op in ops.items():
+                op_weights = self._ops_weights[op_name]
+                for weight_val in op_weights.values():
+                    if weight_val.trainable:
+                        if op.trainable:
+                            trainable_ops_weights_on.append(weight_val)
+                        else:
+                            trainable_ops_weights_off.append(weight_val)
```

Comment:
IMHO: Removed one level of nesting

```suggestion
                    if not weight_val.trainable:
                        continue
                        
                    if op.trainable:
                        trainable_ops_weights_on.append(weight_val)
                    else:
                        trainable_ops_weights_off.append(weight_val)
```